### ISSUE TITLE
SC3043?: local with options is bashism

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3395]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC3043?
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/dash
#shellcheck disable=SC2034
f() {
        local -r var=foo
}
```

#### Here's what shellcheck currently says:

nothing

#### Here's what I wanted or expected to see:

dash does not support `-r` and other options:

```console
$ f() { local -r var=foo; }
$ f
dash: 1: local: -r: bad variable name
```

```
-:4:2: error: In dash, 'local' accepts no option. [SC3043]
```
